src/view/dataElements/xdmObject.jsx

```diff
@@ -79,45 +113,102 @@ const XdmObject = ({ initInfo, formikProps, schema }) => {
 XdmObject.propTypes = {
   initInfo: PropTypes.object.isRequired,
   formikProps: PropTypes.object.isRequired,
-  schema: PropTypes.object
+  schemasMeta: PropTypes.array,
+  selectedSchemaMeta: PropTypes.object,
+  setSelectedSchemaMeta: PropTypes.func
 };
 
 const XdmExtensionView = () => {
-  const [schema, setSchema] = useState();
+  const [schemasMeta, setSchemasMeta] = useState();
+  const [selectedSchemaMeta, setSelectedSchemaMeta] = useState();
 
   return (
     <ExtensionView
       getInitialValues={({ initInfo }) => {
-        return fetchSchema({
-          edgeConfigId: initInfo.extensionSettings.instances[0].edgeConfigId,
+        // TODO: If we've already loaded the schemas, then don't load the
+        // schemas again. Just call getInitialValues with the newly
+        // selected schema and return the result.
+        return fetchSchemaList({
           orgId: initInfo.company.orgId,
           imsAccess: initInfo.tokens.imsAccess
-        }).then(fetchedSchema => {
-          setSchema(fetchedSchema);
+        })
+          .then(_schemasMeta => {
+            setSchemasMeta(_schemasMeta);
 
-          if (!fetchedSchema) {
-            return {};
-          }
+            if (!_schemasMeta.length) {
+              return {};
+            }
 
-          const initialValues = getInitialValues({
-            settings: initInfo.settings,
-            schema: fetchedSchema
-          });
+            // TODO: don't initialize form state if the schema doesn't match
+            let existingSchema;
+            if (initInfo.settings && initInfo.settings.schema) {
+              existingSchema = findSchemaMetaById({
+                $id: initInfo.settings.schema.id,
+                schemasMeta: _schemasMeta
+              });
+            }
+
+            let schemaMeta = _schemasMeta[0];
+            if (existingSchema) {
+              schemaMeta = existingSchema;
+            }
+
+            setSelectedSchemaMeta(schemaMeta);
 
-          return initialValues;
-        });
+            return fetchSchema({
+              orgId: initInfo.company.orgId,
+              imsAccess: initInfo.tokens.imsAccess,
+              schemaMeta
+            });
+          })
+          .then(schema => {
+            const initialValues = getInitialFormState({
+              value: (initInfo.settings && initInfo.settings.data) || {},
+              schema
+            });
+
+            return initialValues;
+          });
       }}
       getSettings={({ values }) => {
         return {
           schema: {
-            id: schema.$id,
-            version: schema.version
+            id: selectedSchemaMeta.$id,
+            version: selectedSchemaMeta.version
           },
           data: getValueFromFormState({ formStateNode: values }) || {}
         };
       }}
       validate={validate}
-      render={props => <XdmObject {...props} schema={schema} />}
+      render={options => {
+        const onSchemaMetaSelected = _schemaMeta => {
```

Comment:
It might be worth adding a TODO for handling race conditions when the user selects a schema then selects another schema before the first schema is done loading. We either need to cancel the previous request or disable the Schema dropdown while the schema is loading.